### PR TITLE
fix: [MEW-2032] report MEW API fetch failures with a descriptive message

### PR DESCRIPTION
**Pull request by renerpdev**
## What was wrong

When a MEW API request failed persistently on the Home view (a network error, or a non-OK HTTP status after 3 retries), the app logged it to Sentry as **"Error: No error message"** — with no endpoint, status, or any useful detail. All 484 of these production events collapsed into a single, unactionable Sentry issue, so we couldn't tell which API call was failing or why.

**Root cause:** the fetch library (`@vueuse/core`) creates its error object from the HTTP "status text", which is **always empty over HTTP/2** (how our site and API are served). So the error we sent to Sentry had a blank message. Ironically, a friendly message was computed one line later — but only for the on-screen error, never for Sentry.

## What changed

- Added a small, fully unit-tested helper that builds a **meaningful, never-empty error message** from the best available signal: the API's own error message → HTTP status text → HTTP status code → "network error".
- When reporting to Sentry, we now also attach the **endpoint URL and status** as tags and group failures **by HTTP status** (e.g. all 404s together, all 500s together, network errors together) instead of lumping everything into one bucket.
- No user-facing behavior changes: the on-screen error message and the retry logic are untouched. This only improves what we record in Sentry (production only).

## How to test

> Note: the Sentry reporting path only runs in **production builds** (`!isDevMode`), so a local dev smoke test verifies **no regression** rather than the enriched Sentry payload directly (which is fully covered by unit tests).

1. Open the dev server and load the Home view connected with a wallet (e.g. mnemonic), on any network.
2. Confirm the app loads normally, balances/prices fetch, and there are no console errors from the fetch/retry path.
3. (Optional) Simulate a failing MEW API request (e.g. block the API host in devtools) and confirm the app still shows its normal error state and stays responsive — no crash, retries still happen.

**Expected:** Home behaves exactly as before; the fetch/retry/error-handling flow is unchanged.

Sentry issue: https://myetherwallet-inc.sentry.io/issues/APP-MEW-WEB-1AE

## Summary by CodeRabbit

* **Bug Fixes**
  * Improved error messages for API and network failures, including clearer HTTP status details.
  * Preserved meaningful original error information when available.
  * Added more consistent categorization of failures with structured fingerprinting and tagging.
  * Request diagnostics now exclude query parameters from identifying labels while retaining full URL details for investigation.
* **Tests**
  * Added unit coverage to ensure error message selection, fingerprinting, and diagnostic payload construction behave consistently across edge cases.

### CODE REVIEW

File: src/composables/mewApiFetchError.ts [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+ * message; otherwise wraps the built message in a fresh Error.
+ */
+export const describeMewApiFetchError = (
+  ctx: MewApiFetchErrorContext,
+): MewApiCapturedError => {
+  const message = buildMewApiErrorMessage(ctx)
+  const error =
+    ctx.error instanceof Error && ctx.error.message
+      ? ctx.error
+      : new Error(message)
```
[the remark this answers is not in the document]
Comment (renerpdev) [reply src/composables/mewApiFetchError.ts]: Done in b64fe77f06 — the synthesized Error now preserves the original as `cause` so Sentry keeps the original stack. The `Error(message, { cause })` constructor option isn't in this project's TS lib (TS2554), so it's attached right after construction instead; the spec verifies `captured.error.cause === original`.

**Review by gamalielhere — APPROVED** [review]
